lib/output/writer/s3.go

```diff
@@ -162,9 +174,17 @@ func (a *AmazonS3) WriteWithContext(wctx context.Context, msg types.Message) err
 			contentEncoding = aws.String(ce)
 		}
 
+		// Prepare tags.
+		tags := []string{}
+		for k, v := range a.tags {
+			tags = append(tags, k+"="+v.String(i, msg))
```

Comment:
I might be wrong as I haven't read up on S3 tags but it looks like we're building a url query string here, so it's probably worth escaping the keys and values with https://golang.org/pkg/net/url/#QueryEscape

Reply:
This is fixed, great catch 👍